### PR TITLE
Added knowledge check to html_css_basics.md

### PR DESCRIPTION
**Pull request by aaron-kidwell**
- [ X] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/master/CONTRIBUTING.md).

#### 1.Describe the changes made:

Added knowledge check to html_css_basics.md

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

#22259

### CODE REVIEW

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
@aaron-kidwell Wow, this lesson is full of a lot of information! In looking closer at this lesson, I think it would be good to re-visit more of the Learning Outcomes in this Knowledge Check. I realize that repeating similar information may seen redundant, but in this case, that is ultimately the purpose of adding the Knowledge Check. 

Could you please add any of the last 7 Learning Outcome topic that is not already in the Knowledge Check? If possible, re-word them, even if you can only slightly re-word it.

In addition, here a few wording changes:
- For accessibility in HTML, what is the attribute used to describe an image (on screen readers or if it fails to load)?
- For a responsive website, should it be designed mobile-first or desktop-first?"
- What is a div and how are they used?

**Comment by aaron-kidwell** [comment]
Changes made, please let me know if any additional changes or corrections are needed, thanks!

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
Thanks for making these changes. Part of my requested changes got lost in the middle of my previous review. 

Could you please add the topics in the last 7 Learning Outcomes that are not already in the Knowledge Check? If possible, re-word them, even if you can only slightly re-word it.

What is the “query string” in a URL and what does it do?
What is the difference between “pixels” and “ems”?
How do CSS styles for a particular element get inherited? i.e. how does an element get its “default” styles?
What are two CSS attributes you can change to push an element around on the page?
What are the three different ways to include CSS in your project or use CSS to style a particular element?
What is the “default stylesheet” or “user agent stylesheet”?
Why use a CSS reset file?

**Comment by aaron-kidwell** [comment]
No problem, the last seven of the learning outcomes have been combined with knowledge checks and some reworded, if any wording needs changing let me know, thanks.

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
Thanks for taking the time to make these changes. I have two more suggestions. 

Beside the typo & the extra period, I am unsure what the answer to "What are the twon main typography groups in CSS?. " is. If is is font & text, I would re-word it to be:
What are the two main groups of CSS properties that control typography style?

Another typo for "inheritence". Plus, I think people will be a little confused with the wording non-inheritance in "How does inheritence/non-inheritence work, i.e. how does an element get its “default” styles?", so I would leave out that word and explicitly say CSS styles and end up with something like this:
How does inheritance work for CSS styles, i.e. how does an element get its “default” styles?

**Comment by rlmoser99** [comment]
@aaron-kidwell Since it has almost been 2 weeks, I wanted to check and see if you are able to make the recommended changes, or would you like for me to go ahead and make the changes?

**Comment by aaron-kidwell** [comment]
I have been busy between school and work so sorry! I am making recommended changes asap tonight, thanks!

**Comment by rlmoser99** [comment]
@aaron-kidwell No problem, as I understand that life can get busy. Thanks for the quick reply!

**Comment by aaron-kidwell** [comment]
Thanks for working with my timelines, changes have been pushed. please let me know if there are issues and I will fix immediately.

File: foundations/the_front_end/html_css_basics.md [hunk 1]
```diff
@@ -48,3 +48,22 @@ This section contains helpful links to other content. It isn't required, so cons
...
+* For accessibility in HTML, what is the attribute used to describe an image (on screen readers or if it fails to load)?
+* What is the difference between CSS Grid and Flexbox?
+* For a responsive website, should it be designed mobile-first or desktop-first?
+* Describe the components of the CSS Box Model.
+* In CSS, what is a breakpoint?
+* What is a div and how are they used?
+* What are the two main groups of CSS properties that control typography style?
+* What is the “query string” in a URL and what does it do?
+* What is the difference between “pixels” and “em”?
+* What are the two main groups of CSS properties that control typography style?
```
Comment (rlmoser99) [inline foundations/the_front_end/html_css_basics.md:65]: Looks like this line got added twice. Do you remember what you meant to change here?

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
Let me know if you don't remember what change you meant to do here & I can look deeper at this lesson. Thanks!

**Comment by aaron-kidwell** [comment]
Correction made!

**Review by rlmoser99 — APPROVED** [review]
🥳  Yes, it looks perfect! Thank you so much for taking the time to make this addition to this lesson!

**Pull request merged**